app/mempool/senders_with_nonce.go

```diff
@@ -0,0 +1,69 @@
+// TODO: use with signer extractor once available https://github.com/zeta-chain/node/issues/2156
+
+package mempool
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+)
+
+// GetSendersWithNonce is used to extract sender and nonce information txs
+// if tx is ethermint, it is extracted using from and nonce field
+// if it's cosmos tx, default cosmos way using signatures is used
+func GetSendersWithNonce(tx sdk.Tx) ([]SenderWithNonce, error) {
+	const extensionOptionsEthereumTxTypeURL = "/ethermint.evm.v1.ExtensionOptionsEthereumTx"
+	if txWithExtensions, ok := tx.(authante.HasExtensionOptionsTx); ok {
+		opts := txWithExtensions.GetExtensionOptions()
```

Comment:
Nit: but it makes it more readable if we have two functions 
`getSendersWithNonceEthermint`  && `getSendersWithNonceCosmos` .